src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -413,19 +413,34 @@ const buildWhereClause = ({ schema, query, index }) => {
 
     if (fieldValue.$geoWithin && fieldValue.$geoWithin.$polygon) {
       const polygon = fieldValue.$geoWithin.$polygon;
-      if (!(polygon instanceof Array)) {
-        throw new Parse.Error(
-          Parse.Error.INVALID_JSON,
-          'bad $geoWithin value; $polygon should contain at least 3 GeoPoints'
-        );
-      }
-      if (polygon.length < 3) {
+      let points
+      if (typeof polygon === 'object' && polygon.__type === 'Polygon') {
+        if (!polygon.coordinates && polygon.coordinates.length < 3) {
```

Comment:
!polygon.coordinates || polygon.coordinates.length < 3

Using `OR` here instead of `AND` passes test. You might want to do the same with mongo and check your logic